### PR TITLE
CONSOLE-3243: Rename "master" to "control plane node" in node pages

### PR DESCRIPTION
**Pull request by tvu20**
Addresses https://issues.redhat.com/browse/CONSOLE-3243

Changes all string occurences of "master" in the console to "control plane node".

### CODE REVIEW

**Comment by jcaianirh** [comment]
QE Approver:
/assign @yapei

Docs Approver:
/assign @opayne1

PX Approver:
/assign @RickJWagner

Console Approver:
/assign @rhamilto

**Review by RickJWagner — APPROVED** [review]
8 files reviewed, good.

**Comment by rhamilto** [comment]
Nice work, @tvu20!  Also need to include updates to Cluster Settings:

* https://github.com/openshift/console/blob/48991b57ac9e73f352fefd3f6ce7402e7da74403/frontend/public/module/k8s/cluster-settings.ts#L309 becomes `Master = 'Control plane',`
* https://github.com/openshift/console/blob/48991b57ac9e73f352fefd3f6ce7402e7da74403/frontend/public/components/cluster-settings/cluster-settings.tsx#L619-L624 becomes 

```
  const mcpName = machineConfigPool?.metadata?.name; 
  const machineConfigPoolIsEditable = useAccessReview({
    group: MachineConfigPoolModel.apiGroup,
    resource: MachineConfigPoolModel.plural,
    verb: 'patch',
    name: mcpName,
  });
```
* https://github.com/openshift/console/blob/48991b57ac9e73f352fefd3f6ce7402e7da74403/frontend/public/components/cluster-settings/cluster-settings.tsx#L648 becomes `to={`/k8s/cluster/nodes?rowFilter-node-role=${mcpName}`}`

It strikes me as a little odd that we're changing "master" to "control plane", but the underlying resource labels aren't changing.  So we still have "master" in the UI.  For example,

https://user-images.githubusercontent.com/895728/183096759-0aeb6d02-9b32-4ae5-a43b-2aae66246583.mov

Screen Shot 2022-08-05 at 10 21 50 AM

Screen Shot 2022-08-05 at 10 21 18 AM
Screen Shot 2022-08-05 at 10 21 30 AM

Do we need to engage UXD here so as to avoid creating confusion for users?  cc:  @megan-hall

File: frontend/packages/console-app/src/components/nodes/NodesPage.tsx
Comment (rhamilto) [inline frontend/packages/console-app/src/components/nodes/NodesPage.tsx:515]:
```suggestion
            title: t('console-app~Control plane'),
```

**Comment by rhamilto** [comment]
@tvu20, I think this probably merits a wider discussion.  Unless the resource names (e.g., /k8s/cluster/machineconfiguration.openshift.io~v1~MachineConfigPool/master) and associated labels are changing, there is a limit to how far we can take these changes, and I fear it may create confusion on the part of end users.  We probably need to revisit the intent of the story with @jhadvig and the stakeholder.

File: frontend/public/module/k8s/cluster-settings.ts
Comment (rhamilto) [inline frontend/public/module/k8s/cluster-settings.ts:300]: This is problematic for https://github.com/openshift/console/blob/f4247fbf08785207f9ad57929f823167eb071c8c/frontend/public/module/k8s/cluster-settings.ts#L313

**Comment by yapei** [comment]
@XiyunZhao  is working on this PR

**Comment by XiyunZhao** [comment]
Hi @tvu20, based on current changes, I opened two bugs for this new feature, could you help to identify whether they need to update or not? 
1. The name of "master" on "Filter by node type" dropdown list on cluster utilization on the overview page is not changed 
    https://bugzilla.redhat.com/show_bug.cgi?id=2116608
overview_nodetype
2. After the changes, duplicate role name will be listed on "Nodes" page which is incorrect
    https://bugzilla.redhat.com/show_bug.cgi?id=2116625
nodes-controlplan

**Comment by tvu20** [comment]
Hi @XiyunZhao, I have just pushed a new commit that addresses bug 1. I am not able to replicate the second bug on my end.

**Comment by XiyunZhao** [comment]
@tvu20 sorry for bug 2, after checking, it is not a duplicate value.
I would like to ask what's the difference between "control plane" with "control-plane"? Will it mislead customers?

**Comment by jhadvig** [comment]
@opayne1 could you please elaborate on the [previous ](https://github.com/openshift/console/pull/11927#issuecomment-1214894187)question? Thanks !

**Comment by opayne1** [comment]
@jhadvig @XiyunZhao In our documentation, we use "control plane" with no hyphen. See the [Control plane architecture docs](https://docs.openshift.com/container-platform/4.10/architecture/control-plane.html) for an example. We also have it as "control planes" no hyphen in our [documentation contributor guidelines](https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#using-conscious-language). Adding the docs-approved label as this PR looks good to me.

**Comment by XiyunZhao** [comment]
> @tvu20, I think this probably merits a wider discussion. Unless the resource names (e.g., /k8s/cluster/machineconfiguration.openshift.io~v1~MachineConfigPool/master) and associated labels are changing, there is a limit to how far we can take these changes, and I fear it may create confusion on the part of end users. We probably need to revisit the intent of the story with @jhadvig and the stakeholder.

@tvu20 Is there any conclusion for this comment? Could you help to add some details to features description?

**Comment by jhadvig** [comment]
> @tvu20, I think this probably merits a wider discussion. Unless the resource names (e.g., /k8s/cluster/machineconfiguration.openshift.iov1MachineConfigPool/master) and associated labels are changing, there is a limit to how far we can take these changes, and I fear it may create confusion on the part of end users. We probably need to revisit the intent of the story with @jhadvig and the stakeholder.

I agree with @rhamilto. We should revisit the story since it can lead to confusion between Nodes, MachineConfig and MachineConfigPool resources, which are still using the legacy `master` naming. Until that is solved, this change would only lead to user confusion. We should be consistent with the naming.

/hold

**Comment by spadgett** [comment]
I'm in favor of doing what we can to get away from this terminology, but I agree this will create confusion since the actual names and labels of the resources are still `master`. You'll need to know the real value when using the CLI, writing label selectors, or editing the resource YAML. Even in the UI, we would show "control plane" in some places and "master" in others as @rhamilto points out.

Have we considered fixing this at the platform level and changing the actual resource names? (I'm sure this is easier said than done since names are immutable, this is arguably API breaking, and we probably don't want to drift from upstream...)

@deads2k @sdodson Has this come up before?

**Comment by sdodson** [comment]
> @deads2k @sdodson Has this come up before?

Poking around to see what I can find. This seems to have happened upstream via kubeadm(https://github.com/kubernetes/kubeadm/issues/2200) but we won't pick those changes up in OCP as we don't leverage kubeadm.

**Comment by spadgett** [comment]
@sdodson has pointed out that we should at least be able to use a new node role in 4.12: https://bugzilla.redhat.com/show_bug.cgi?id=1995858 (thanks!)

**Comment by spadgett** [comment]
OK, here is the upstream guidance:

https://github.com/kubernetes/community/blob/master/sig-architecture/naming/recommendations/001-master-control-plane.md

Based on this, I'd suggest 

1. We use "control plane" or "control plane nodes" in text descriptions, help, etc. rather than "master"
2. We use the `node-role.kubernetes.io/control-plane` role instead of `node-role.kubernetes.io/master`
3. We only use `master` when displaying a name of a resource

**Comment by jcaianirh** [comment]
@spadgett @rhamilto So based on Sams comment above, is this PR / story still needed or is this change a docs only or help only change? If this is a code change, can we push this to 4.13?

**Comment by rhamilto** [comment]
> https://github.com/kubernetes/community/blob/master/sig-architecture/naming/recommendations/001-master-control-plane.md

Yes, we'll still need console code changes, most of which are in this PR, but we need a few tweaks based on the guidance.

**Comment by rhamilto** [comment]
Replaced by https://github.com/openshift/console/pull/12209

**Pull request closed without merging**
